### PR TITLE
Add Repository Integrated check

### PR DESCRIPTION
**Pull request by rocktavious**
## Issues

Add repo integrated check

## Tophatting

Create
```bash
  # opslevel_check_repository_integrated.example will be created
  + resource "opslevel_check_repository_integrated" "example" {
      + category     = "Z2lkOi8vb3BzbGV2ZWwvQ2F0ZWdvcnkvOTY1"
      + description  = (known after apply)
      + enabled      = true
      + id           = (known after apply)
      + last_updated = (known after apply)
      + level        = "Z2lkOi8vb3BzbGV2ZWwvTGV2ZWwvNTIw"
      + name         = "foo"
    }

```

Update
```bash
  # opslevel_check_repository_integrated.example will be updated in-place
  ~ resource "opslevel_check_repository_integrated" "example" {
        id           = "Z2lkOi8vb3BzbGV2ZWwvQ2hlY2tzOjpIYXNSZXBvc2l0b3J5LzI0MTc3"
      + last_updated = (known after apply)
      ~ name         = "foo" -> "foobar"
        # (4 unchanged attributes hidden)
    }

```

Delete
```
  # opslevel_check_repository_integrated.example will be destroyed
  # (because opslevel_check_repository_integrated.example is not in configuration)
  - resource "opslevel_check_repository_integrated" "example" {
      - category    = "Z2lkOi8vb3BzbGV2ZWwvQ2F0ZWdvcnkvOTY1" -> null
      - description = "Verifies that the service has a repository integrated." -> null
      - enabled     = true -> null
      - id          = "Z2lkOi8vb3BzbGV2ZWwvQ2hlY2tzOjpIYXNSZXBvc2l0b3J5LzI0MTc3" -> null
      - level       = "Z2lkOi8vb3BzbGV2ZWwvTGV2ZWwvNTIw" -> null
      - name        = "foobar" -> null
    }

```

### CODE REVIEW

File: opslevel/resource_opslevel_check_repository_integrated.go [hunk 1]
```diff
@@ -1,59 +1,196 @@
...
+	model.Category = types.StringValue(string(check.Category.Id))
+	model.Enabled = types.BoolValue(check.Enabled)
+	model.EnableOn = types.StringValue(check.EnableOn.Time.Format(time.RFC3339))
+	model.Filter = types.StringValue(string(check.Filter.Id))
+	model.Id = types.StringValue(string(check.Id))
+	model.Level = types.StringValue(string(check.Level.Id))
+	model.Name = types.StringValue(check.Name)
+	model.Notes = types.StringValue(check.Notes)
+	model.Owner = types.StringValue(string(check.Owner.Team.Id))
+	model.LastUpdated = timeLastUpdated()
```
Comment (taimoorgit) [inline opslevel/resource_opslevel_check_repository_integrated.go:57]: Needs to be moved

Comment (taimoorgit) [reply opslevel/resource_opslevel_check_repository_integrated.go:57]: Also, we should use the terraform type conversion functions in the `NewResourceModel` functions here as part of standardization -- non-blocking.

File: opslevel/resource_opslevel_check_repository_integrated.go [hunk 2]
```diff
@@ -1,59 +1,196 @@
...
+	// Read Terraform plan data into the planModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	enabledOn, err := iso8601.ParseString(planModel.EnableOn.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("error", err.Error())
```
Comment (taimoorgit) [inline opslevel/resource_opslevel_check_repository_integrated.go:87]: Need `return` after this statement

**Review by taimoorgit — APPROVED** [review]
LGTM -- 1 suggestion, 1 missing return statement

File: opslevel/resource_opslevel_check_repository_integrated.go (hunk 1 above)
Comment (rocktavious) [reply opslevel/resource_opslevel_check_repository_integrated.go:57]: done

File: opslevel/resource_opslevel_check_repository_integrated.go (hunk 2 above)
Comment (rocktavious) [reply opslevel/resource_opslevel_check_repository_integrated.go:87]: will fix in post

**Pull request merged**